spec/ruby/language/block_spec.rb

```diff
@@ -983,3 +983,77 @@ def a; 1; end
     end
   end
 end
+
+describe "Anonymous block forwarding" do
+  ruby_version_is "3.1" do
+    it "forwards blocks" do
+      eval <<-EOF
+          def b(&); c(&) end
+          def c(&); yield :non_null end
```

Comment:
Is the `(&)` needed here? It seems superfluous and not relevant for this example

Reply:
If I remove the explicit param, it's no different than the 'works when it's the only declared parameter' spec. Given MRI 3.1 shipped with a broken parser, I wanted to ensure the syntax worked the way it is expected to.

Reply:
Then we need a better description for this example, to hint at what it is testing.

Reply:
It's testing that a block can be forwarded. As noted in the PR, I took it from the MRI tests: https://bugs.ruby-lang.org/projects/ruby-master/repository/git/revisions/4adb012926f8bd6011168327d8832cf19976de40/diff/test/ruby/test_syntax.rb.

Reply:
In general ruby/spec examples try to make sense and describe what they test. MRI tests OTOH...
So concretely here I would move `works when it's the only declared parameter` first.
And then this one as second example, and rename to `allows a & parameter even if it is not used`, which then also nicely completes the example below.

Reply:
Neither of those descriptions are what I intended. But, I did port them from MRI so if I misunderstood the author's original intentions, so be it. The first one is a basic test that forwarding works. The second spec is verifying behavior with a variety of different argument forms. They were taken from two different MRI commits, the first one having an insufficient number of tests that led to an unexpected parse error when explicit kwargs are used.

We can just delete the first test if you prefer. But, given the rules around when an anonymous block must be declared and how it can be forwarded, I think it makes sense to test the different forms. The function declaration and the value passing are two different paths in the parser.

Given our difference of opinion on what's being tested or what should be a Ruby Spec vs an MRI test, please feel free to push your changes directly to the branch. We're in a bit of an odd situation where MRI tests exist, but we can't import them yet, and no Ruby Specs exist at all for this functionality. Normally, I find the specs and the MRI tests complement each other well so I try not to just port the MRI tests, but lacking any other way to test the parser changes made to support anonymous block forwarding, I went with the lower-level comprehensive checks the MRI tests perform.

Reply:
Maybe it could be more descriptive than "forwards blocks"? E.g. "forwards blocks without a variable name" or "forwards blocks using a shorthand syntax".

Reply:
Okay, this could be my own misunderstanding on naming nested specs. I was joining the name with the outer describe block so it'd be "Anonymous block forwarding forwards blocks". The "anonymous" part covers not having a variable name. I'll rename it to "Anonymous blocks forwarding forwards blocks to other functions that formally declare anonymous blocks."